frontend/app/(defaultLayout)/Troubleshooting/[device]/page.tsx

```diff
@@ -19,7 +20,12 @@ export default function Page({ params, searchParams }: PageProps) {
    ensureFlag();
 
    const pageProps = getPageProps({ params, searchParams });
-   return <PageTemplate {...pageProps} />;
+   return (
+      <>
+         <Header {...pageProps} />
+         <ServerHeader {...pageProps} />
```

Comment:
Why is there a server header component vs a client component? This seems very unnecessary

Reply:
I should clarify that the component was still rendered server side even with a use client statement

Reply:
The component does not have any additional functional purpose.
I intentionally created a duplicated component just to underline the difference between a server component and a server rendered client component. 
It might be not so obvious for anyone, since RSC are a pretty new topic, so I'll use your question to build a little bit more context for anyone that might end up reading this.

When Next.js receives a request, it performs a complete server-rendering of the component hierarchy. This process involves both server components and client components.
The real difference from RSC and RCC lies in the payload that is sent to the client: if you inspect the bundle, you would notice how the `ServerHeader` component is not even included in the bundle, while the `Header` component is.
Even though the impact of a single component might be tiny, we have to consider that also dependencies might or might not be included according to a component being a RSC or a RCC.

Therefore - whenever possible - we should aim to use client components only when needed and especially on leaf nodes of the component hierarchy.

If you consider the current ecommerce UI, several parts if not most parts could be made by RSC.
The real blocker at the moment is Chakra UI, that is not compatible with RSC, due to the fact that handles state internally. So, me and @dhmacs are probably going to iterate a bit more on this hello-world-app-router branch to test alternatives that could allow a wider adoption of RSC (Tailwind css)

<img width="1720" alt="Screenshot 2023-09-08 at 11 21 52" src="https://github.com/iFixit/react-commerce/assets/7805759/e103ab49-6e75-4325-9ada-d684dac73d66">
<img width="1727" alt="Screenshot 2023-09-08 at 11 22 00" src="https://github.com/iFixit/react-commerce/assets/7805759/7ccf8e76-8cba-4dc0-9d45-fa1f011f04ac">